### PR TITLE
TINY-9161: Upgrade to the latest rollup-plugin-dts version

### PR DESCRIPTION
**Pull request by lnewson**
Related Ticket: TINY-9161

Description of Changes:
* Upgrade to the latest `rollup-plugin-dts` version to pull in a fix for generating invalid generics and bumped relevant dependencies.

Pre-checks:
* [x] Changelog entry added
* [x] package.json version bumped (if first change of new major/minor)
* [x] Tests have been added (if applicable)

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -28,15 +28,15 @@
   "dependencies": {
     "acorn": "^8.0.0",
     "escodegen": "^2.0.0",
-    "magic-string": "^0.25.4",
+    "magic-string": "^0.26.2",
     "resolve": "^1.11.0",
-    "rollup": "^2.48.0",
-    "rollup-plugin-dts": "^3.0.2",
+    "rollup": "^2.77.0",
```
Comment (lnewson) [inline package.json:33]: I upgraded these 2 to match the minimum deps for the rollup DTS plugin.

**Comment by lnewson** [comment]
Somehow have got a TypeScript compilation error along the way. Moving back to a draft while I fix.

**Comment by lnewson** [comment]
Found the cause... it's because Rollup 2.78.0 introduced a breaking change whereby a plugin hook can either be a function or an object.

File: CHANGELOG.md [hunk 2]
```diff
@@ -6,6 +6,14 @@ and this project adheres to [Semantic Versioning](http://semver.org/spec/v2.0.0.
 
 ## Unreleased
 
+## 4.6.0 - 2022-09-15
+
+### Changed
+- Upgraded the minimum rollup and rollup-plugin-dts versions.
+
+### Fixed
+- Type generics could be incorrectly when .d.ts files are generated.
```
Comment (ltrouton) [inline CHANGELOG.md:15]:
```suggestion
- Type generics could be incorrect when .d.ts files are generated.
```

**Review by ltrouton — APPROVED** [review]

**Review by spocke — APPROVED** [review]

**Pull request merged**